src/dev/run_extract_default_translations.js

```diff
@@ -0,0 +1,343 @@
+    ids.push(...parseConditionalOperatorAST(subTree.alternate));
+  }
+
+  return ids;
+}
+
+function parseConditionalOperatorExpression(expression) {
+  let ids = [];
+
+  traverse(parse(expression), {
+    enter(path) {
+      if (babelTypes.isDirectiveLiteral(path.node)) {
+        ids = [path.node.value];
+        path.stop();
+      } else if (babelTypes.isConditionalExpression(path.node)) {
+        ids = parseConditionalOperatorAST(path.node);
+        path.stop();
+      }
+    },
+  });
+
+  return ids;
+}
+
+function parseFilterObjectExpression(expression) {
+  let defaultMessage = '';
+
+  const filterObjectAST = parse(`+${expression}`);
+  traverse(filterObjectAST, {
+    enter(path) {
+      if (babelTypes.isObjectExpression(path.node)) {
+        path.node.properties.forEach(property => {
+          if (isPropertyWithKey(property, defaultMessageKey) && babelTypes.isStringLiteral(property.value)) {
+            defaultMessage = property.value.value;
+          }
+        });
+      }
+    }
+  });
+
+  return defaultMessage;
+}
+
+function getFilterMessages(htmlContent) {
+  const expressions = (htmlContent.match(angularExpressionRegEx) || [])
+    .filter(expression => expression.indexOf('|') !== -1)
+    .filter(expression => expression.indexOf('i18n') !== -1)
+    .map(expression => expression.slice(2, -2).trim());
+
+  const defaultMessagesMap = new Map();
+
+  expressions.forEach(expression => {
+    const [idsExpression] = expression.match(/[\s\S]+(?=\|[\s]*i18n)/g) || [];
+    const [filterObjectExpression] = expression.match(/(?<=\|[\s]*i18n[\s]*:)[\s\S]+/g) || [];
+
+    if (!filterObjectExpression || !idsExpression) {
+      throw new Error('Angular.js expression syntax error');
+    }
+
+    const messagesIds = parseConditionalOperatorExpression(idsExpression);
+    const messageValue = parseFilterObjectExpression(filterObjectExpression);
+
+    messagesIds.forEach(id => {
+      addMessageToMap(defaultMessagesMap, id, messageValue);
+    });
+  });
+
+  return defaultMessagesMap;
+}
+
+function getMessagesIdsFromHTMLElement(attributeValue) {
+  if (!attributeValue) {
+    return [];
+  }
+
+  const [expression] = attributeValue.match(angularExpressionRegEx) || [];
+
+  if (expression) {
+    return parseConditionalOperatorExpression(expression.slice(2, -2).trim());
+  }
+
+  return [attributeValue];
+}
+
+function getDirectiveMessages(htmlContent) {
+  const defaultMessagesMap = new Map();
+  const document = jsdom(htmlContent, {
+    features: { ProcessExternalResources: false },
+  }).defaultView.document;
+
+  Array.prototype.slice.call(document.getElementsByTagName('*'))
+    .filter(element => element.hasAttribute('i18n-id'))
+    .forEach(element => {
+      const messagesIds = getMessagesIdsFromHTMLElement(element.getAttribute('i18n-id'));
+      const messageValue = element.getAttribute('i18n-default-message') || '';
+
+      if (messagesIds.length > 0 && !messageValue) {
```

Comment:
__question:__ Do we really need this part `messagesIds.length > 0 && ` of the condition?

Reply:
It seems we can throw two different errors here:
* when we can't find default message
* when we can't parse message id